### PR TITLE
[WIP] Fix XDeathMaxLifetimeProcessor when time is \AMQPTimestamp

### PR DESCRIPTION
**Pull request by instabledesign**
|**Q**|**A**|
| ------------- |:-------------:|
| Branch? | master|
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | no |

### CODE REVIEW

File: tests/Swarrot/Processor/XDeath/XDeathMaxLifetimeProcessorTest.php [hunk 1]
```diff
@@ -121,7 +121,22 @@ public function messageProvider()
...
+                            'x-death' => [
+                                ['time' => new \AMQPTimestamp(time() - 50)],
+                            ],
+                        ],
+                    ]
+                ),
+            ];
+        }
+
+        if (false && class_exists('PhpAmqpLib\Wire\AMQPArray') && class_exists('PhpAmqpLib\Wire\AMQPTable')) {
```
Comment (odolbeau) [inline tests/Swarrot/Processor/XDeath/XDeathMaxLifetimeProcessorTest.php]: `if (false)` ?

File: src/Swarrot/Processor/XDeath/XDeathMaxLifetimeProcessor.php [hunk 2]
```diff
@@ -117,8 +117,7 @@ private function handleException($exception, Message $message, array $options)
                 );
             } elseif (isset($queueXDeathHeader['time'])) {
                 $xDeathTimestamp = $queueXDeathHeader['time'];
-                // PhpAmqpLib compatibility
-                if ($xDeathTimestamp instanceof \DateTime) {
+                if (method_exists($xDeathTimestamp, 'getTimestamp')) {
```
Comment (stof) [inline src/Swarrot/Processor/XDeath/XDeathMaxLifetimeProcessor.php]: I think having `$xDeathTimestamp instanceof \DateTime || $xDeathTimestamp instanceof \AMQPTimestamp` might be better to see what is actually supported

**Comment by odolbeau** [comment]
I removed phpstan, you can rebase your branch!

**Comment by instabledesign** [comment]
seems to be good
but php nightly failed to build

**Comment by odolbeau** [comment]
Compilation of the extension have change few months ago. If I remember well, I unsuccessfully tried to repair the build at this time. :/
Anyway, thanks a lot for your contribution! :+1:

**Comment by odolbeau** [comment]
Oops, there's still the `[WIP]` in the commit message... :(

**Pull request merged**